### ISSUE TITLE
Lint to warn about str::as_ptr() usage in ffi call

### ISSUE THREAD

**Issue by antoyo** [Manishearth/rust-clippy#1236]
Hello.
It would be nice to have a warning when using `str::as_ptr()` in a ffi call instead of using `CStr`.

For instance, in the following code:
```rust
unsafe { printf("Hello".as_ptr() as *const _) };
```
should give a warning and propose something like:
```rust
let cstring = CString::new("Hello".as_bytes()).unwrap();
unsafe { printf(cstring.as_ptr()) };
```
Thanks to add this lint.
